python/cudf/cudf/core/multiindex.py

```diff
@@ -117,7 +123,7 @@ def __init__(
 
     @property
     def names(self):
-        return self._names
+        return pd.core.indexes.frozen.FrozenList(self._names)
```

Comment:
Should we handle the typing in `names.setter` to avoid having to possibly copy construct this each time?

Reply:
I believe @trxcllnt was running into something similar to this in concat

Reply:
Yep. I've improved this slightly so that if the provided `names` are non-unique, we set them to be the column names of the `source_data`. This should help with some of the issues wrt MultiIndex names.

Will try to handle typing in `names.setter`.

Reply:
Fixed